main/api/v1/auth.py

```diff
@@ -16,8 +18,13 @@
 @api_v1.resource('/auth/signin/', endpoint='api.auth.signin')
 class AuthAPI(restful.Resource):
   def post(self):
-    username = util.param('username') or util.param('email')
-    password = util.param('password')
+    args = parser.parse({
+        'username': wf.Str(missing=None),
+        'email': wf.Str(missing=None),
+        'password': wf.Str(missing=None),
+      })
```

Comment:
Just simple experiment
``` python
#!/usr/bin/env python
from timeit import timeit
import flask

from webargs.flaskparser import parser
from webargs import fields as wf


app = flask.Flask(__name__)

def param(name, cast=None):
  def switch(case):
    return {
      int: wf.Int(missing=None),
      bool: wf.Bool(missing=None),
      list: wf.DelimitedList(wf.Str(), delimiter=',', missing=[]),
    }.get(case)
  if cast is None:
    cast_ = wf.Str(missing=None)
  else:
    cast_ = switch(cast) or cast
  args = parser.parse({name: cast_})
  return args[name]


@app.route('/args/')
def args_route():
  args = parser.parse({
      'cursor': wf.Str(missing=None),
      'limit': wf.Int(missing=None),
      'order': wf.Str(missing=None),
    })
  return flask.jsonify({'result': [args['cursor'], args['limit'], args['order']]})


@app.route('/args_huge/')
def args_route_huge():
  args = parser.parse({
      'cursor': wf.Str(missing=None),
      'limit': wf.Int(missing=None),
      'order': wf.Str(missing=None),
      'order1': wf.Str(missing=None),
      'order2': wf.Str(missing=None),
      'order3': wf.Str(missing=None),
      'order4': wf.Str(missing=None),
      'order5': wf.Str(missing=None),
      'order6': wf.Str(missing=None),
      'order7': wf.Str(missing=None),
      'order8': wf.Str(missing=None),
      'order8': wf.Str(missing=None),
      'order10': wf.Str(missing=None),
      'order11': wf.Str(missing=None),
    })
  return flask.jsonify({'result': [args['cursor'], args['limit'], args['order']]})


@app.route('/param/')
def param_route():
  cursor = param('cursor')
  limit = param('limit', int)
  order = param('order')
  return flask.jsonify({'result': [cursor, limit, order]})


@app.route('/param_huge/')
def param_route_huge():
  cursor = param('cursor')
  limit = param('limit', int)
  order = param('order')
  order = param('order1')
  order = param('order2')
  order = param('order3')
  order = param('order4')
  order = param('order5')
  order = param('order6')
  order = param('order7')
  order = param('order8')
  order = param('order9')
  order = param('order10')
  order = param('order11')
  return flask.jsonify({'result': [cursor, limit, order]})


def test_1():
  with app.test_request_context('/args/?cursor=123&limit=1&order=email', method='get'):
    args_route()


def test_2():
  with app.test_request_context('/param/?cursor=123&limit=1&order=email', method='get'):
    param_route()


def test_3():
  with app.test_request_context('/args_huge/?cursor=123&limit=1&order=email', method='get'):
    args_route_huge()


def test_4():
  with app.test_request_context('/param_huge/?cursor=123&limit=1&order=email', method='get'):
    param_route_huge()


print timeit(test_1, number=100)
print timeit(test_2, number=100)
print '------------------------'
print timeit(test_3, number=100)
print timeit(test_4, number=100)
```
results:
``` shell
0.128821849823
0.227410078049
------------------------
0.199497938156
0.895273923874
```
therefore call multiple param() is less efficient than call one parse()